doc/topics/the-event-loop-timers-and-nexttick.md

```diff
@@ -0,0 +1,183 @@
+# Overview of the Event Loop, Timers, and `process.nextTick()`
+
+The Following diagram shows a simplified overview of the event loop's
+order of operations.
+
+       ┌───────────────────────┐
+    ┌─>│        timers         │
+    │  └──────────┬────────────┘
+    │  ┌──────────┴────────────┐
+    │  │   pending callbacks   │
+    │  └──────────┬────────────┘      ┌───────────────┐
+    │  ┌──────────┴────────────┐      │   incoming:   │
+    │  │         poll          │<─────│  connections, │
+    │  └──────────┬────────────┘      │   data, etc.  │
+    │  ┌──────────┴────────────┐      └───────────────┘
+    └──│     setImmediate      │
+       └───────────────────────┘
+
+*note: each box will be referred to as a "phase" of the event loop.*
+
+*There is a slight discrepancy between the Windows and the Unix/Linux
+implementation, but that's not important for this demonstration.  The most
+important parts are here.  There are actually seven or eight steps, but the
+ones we care about — ones that Node.js actually uses are these four.*
+
+## timers
+
+This phase executes callbacks scheduled by `setTimeout()` and `setInterval()`.
+
+When you create a timer, you make a call to `setTimeout()`.  Then, when
+the poll phase of the event loop is entered, the number of ms before the 
+soonest timer is to be called is set as the poll's timeout. Meaning the 
+poll phase will return after "timeout" ms.  After that many 
+milliseconds, the `poll` phase will return and wrap back around to the 
+timers phase where those callbacks can be processed.
+
+Take note that the poll phase can only return while idle; execution of 
+a callback is allowed to run to completion, and can cause  unexpected 
+delay running the timer. 
+
+*Note: The `poll` phase technically controls when timers are called due to its
+ability to cause a thread to sit idly without burning CPU in order to stall the
+event loop so the timer can execute.*
+
+## pending callbacks:
+
+This phase executes callbacks for specific types of TCP errors, for example.
```

Comment:
For example if tcp connect receives a `ECONNREFUSED`, some *nix systems want to wait to report the error. This will be queued to execute in the "pending" phase.

@bnoordhuis You may have better examples. I'm just going from libuv source to pull these out. But haven't been able to write a script that demonstrates this precisely.